check/src/kindcheck.rs

```diff
@@ -143,7 +120,7 @@ impl<'a> KindCheck<'a> {
     fn builtin_kind(&self, typ: BuiltinType) -> RcKind {
         match typ {
             BuiltinType::String | BuiltinType::Byte | BuiltinType::Char | BuiltinType::Int |
-            BuiltinType::Float | BuiltinType::Unit => self.type_kind(),
+            BuiltinType::Float | BuiltinType::Unit | BuiltinType::Type => self.type_kind(),
```

Comment:
I think we should at least make a new issue to introduce cumulative universes after this is merged.